frontend/public/components/silence.tsx

```diff
@@ -156,6 +164,164 @@ const SilencesPage_ = props => <MonitoringListPage
 />;
 export const SilencesPage = connectMonitoringPage(connect(monitoringSilencesToProps)(SilencesPage_));
 
+const pad = i => i < 10 ? `0${i}` : i;
+const formatDate = (d: Date): string => `${d.getFullYear()}/${pad(d.getMonth() + 1)}/${pad(d.getDate())} ${pad(d.getHours())}:${pad(d.getMinutes())}:${pad(d.getSeconds())}`;
+
+const toISODate = (dateStr: string): string => {
+  const timestamp = Date.parse(dateStr);
+  return isNaN(timestamp) ? undefined : (new Date(timestamp)).toISOString();
+};
+
+const Text = props => <input {...props} type="text" className="form-control form-control--silence-text" />;
+
+const Datetime = props => <Tooltip content={[<span className="co-nowrap" key="co-timestamp">{toISODate(props.value)}</span>]}>
+  <Text
+    {...props}
+    pattern="\d{4}/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01]) ([01]\d|2[0-3]):[0-5]\d(:[0-5]\d)?"
+    placeholder="YYYY/MM/DD hh:mm:ss"
+  />
+</Tooltip>;
+
+export const CreateSilence = withFallback(connectToURLs(MonitoringRoutes.AlertManager)(class InnerCreateSilence extends SafetyFirst<CreateSilenceProps, CreateSilenceState> {
+  constructor (props) {
+    super(props);
+
+    const now = new Date();
+    const startsAt = formatDate(now);
+    const endsAt = formatDate(new Date(now.setHours(now.getHours() + 2)));
+    const data = {startsAt, endsAt, matchers: [], createdBy: '', comment: ''};
+    this.state = {data, error: undefined, inProgress: false};
+
+    this.addMatcher();
+  }
+
+  /* eslint-disable no-undef */
+  setField = (path: string, v: any): void => {
+    const data = Object.assign({}, this.state.data);
+    _.set(data, path, v);
+    this.setState({data});
+  }
+
+  onFieldChange = (path: string): ((e) => void) => {
+    return e => this.setField(path, e.target.value);
+  }
+
+  onIsRegexChange = (e, i: number): void => {
+    this.setField(`matchers[${i}].isRegex`, e.target.checked);
+  }
+
+  addMatcher = (): void => {
+    this.setField(`matchers[${this.state.data.matchers.length}]`, {name: '', value: '', isRegex: false});
+  }
+
+  removeMatcher = (i: number): void => {
+    const data = Object.assign({}, this.state.data);
+    data.matchers.splice(i, 1);
+    this.setState({data});
+    if (!data.matchers.length) {
+      // All matchers have been removed, so add back a single blank matcher
+      this.addMatcher();
+    }
+  }
+
+  onSubmit = (e): void => {
+    e.preventDefault();
+
+    const alertManagerURL = this.props.urls[MonitoringRoutes.AlertManager];
+    if (!alertManagerURL) {
+      this.setState({error: 'Alertmanager URL not set'});
+      return;
+    }
+
+    this.setState({inProgress: true});
+
+    const body = Object.assign({}, this.state.data);
+    body.startsAt = toISODate(body.startsAt);
+    body.endsAt = toISODate(body.endsAt);
+
+    coFetchJSON.post(`${alertManagerURL}/api/v1/silences`, body)
+      .then(({data}) => {
+        this.setState({error: undefined});
+        history.push(`${SilenceResource.path}/${_.get(data, 'silenceId')}`);
```

Comment:
Is there any chance this could return an ID with characters that need to be encoded in the URL? (I don't know the alert manager API.)

Reply:
The ID characters are just `0-9`, `a-f` and `-`, but I added `encodeURIComponent()` anyway so that we're not relying on that.